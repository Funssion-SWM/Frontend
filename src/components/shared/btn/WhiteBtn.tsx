import { ButtonSize } from '@/types/common';

type Props = {
  text: string;
  onClick: () => void;
  size?: ButtonSize;
  extraStyle?: string;
};

export default function WhiteBtn({
  text,
  onClick,
  size = 'medium',
  extraStyle = '',
}: Props) {
  return (
    <button
      type="button"
      className={`bg-white text-soma-blue-50 border-soma-grey-40 border-[1px] px-3.5 py-2 rounded-3xl transition hover:bg-soma-grey-40 w-fit ${
        {
          big: 'text-sm sm:text-base',
          medium: 'text-xs sm:text-sm',
          small: 'text-[10px] sm:text-xs',
        }[size]
      } ${extraStyle}`}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
