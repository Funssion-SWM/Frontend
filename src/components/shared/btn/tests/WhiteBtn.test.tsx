import renderer from 'react-test-renderer';
import WhiteBtn from '../WhiteBtn';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('BlueBtn2', () => {
  it('renders correctly when size is small', () => {
    const component = renderer.create(
      <WhiteBtn text="text" size="small" onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is medium', () => {
    const component = renderer.create(
      <WhiteBtn text="text" size="medium" onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is big', () => {
    const component = renderer.create(
      <WhiteBtn text="text" size="big" onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('when button clicked', async () => {
    const mockCallBack = jest.fn();

    render(<WhiteBtn text="text" onClick={mockCallBack} />);

    const btn = screen.getByRole('button');
    await userEvent.click(btn);
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
