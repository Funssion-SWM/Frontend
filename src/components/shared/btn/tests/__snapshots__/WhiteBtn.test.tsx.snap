// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlueBtn2 renders correctly when size is big 1`] = `
<button
  className="bg-white text-soma-blue-50 border-soma-grey-40 border-[1px] px-3.5 py-2 rounded-3xl transition hover:bg-soma-grey-40 w-fit text-sm sm:text-base "
  onClick={[Function]}
  type="button"
>
  text
</button>
`;

exports[`BlueBtn2 renders correctly when size is medium 1`] = `
<button
  className="bg-white text-soma-blue-50 border-soma-grey-40 border-[1px] px-3.5 py-2 rounded-3xl transition hover:bg-soma-grey-40 w-fit text-xs sm:text-sm "
  onClick={[Function]}
  type="button"
>
  text
</button>
`;

exports[`BlueBtn2 renders correctly when size is small 1`] = `
<button
  className="bg-white text-soma-blue-50 border-soma-grey-40 border-[1px] px-3.5 py-2 rounded-3xl transition hover:bg-soma-grey-40 w-fit text-[10px] sm:text-xs "
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
